Aries/Aries/Source/Messaging/MessageReceiver.swift

```diff
@@ -45,40 +44,117 @@ public class MessageReceiver{
             case .failure(let error):
                 print("Failed to unpack... \(error)")
             case .success(let data):
-                do {
-                    print(String(data: data, encoding: .utf8)!)
-                    let decoder = JSONDecoder()
-                    let unpackedMessage = try decoder.decode(IndyUnpackedMessage.self, from: data)
-                    let payload = unpackedMessage.message.data(using: .utf8)!
-                    let typeContainer = try decoder.decode(TypeContainerMessage.self, from: payload)
-                    let type = typeContainer.type
-                    self.triggerEvent(type: type, payload: payload, senderVerkey: unpackedMessage.senderVerkey)
-                } catch {
-                    print("Failed to decode...\(error)")
-                }
                 
+                print("Unpacked Message: \(String(data: data, encoding: .utf8)!)")
+                let decoder = JSONDecoder()
+                
+                self.parseDecorators(message: String(data: data, encoding: .utf8)!) { result in
+                    switch result {
+                    case .failure(let error):
+                        print("\(error.localizedDescription)")
+                    case .success(let parsedMessage):
+                        print("Parsed String:   >>>> \(parsedMessage)")
+                        let typeContainer = try! decoder.decode(TypeContainerMessage.self, from: parsedMessage.message.data(using: .utf8)!)
+                        let type = typeContainer.type
+                        print(parsedMessage.message)
+                        self.triggerEvent(type: type, payload: parsedMessage.message.data(using: .utf8)!, senderVerkey: parsedMessage.senderVerkey)
+                    }
+                    return String()
+                }
             }
         }
+    }
+    
+    private func parseDecorators(message: String, completion: @escaping (_ string: Result<IndyUnpackedMessage, Error>)->String) {
```

Comment:
Callback should return void
```suggestion
    private func parseDecorators(message: String, completion: @escaping (_ string: Result<IndyUnpackedMessage, Error>)->Void) {
```